### PR TITLE
Prepare for BMO main branch switch

### PR DESCRIPTION
**Pull request by kashifest**
This PR prepares the prow config for BMO main branch switch. Mainly this PR adds the branch monitoring for BMO in prow config. Currently it monitors two branches main and master. Once main is the default branch, master would be removed.

### CODE REVIEW

File: prow/config/config.yaml [hunk 1]
```diff
@@ -84,8 +84,13 @@ branch-protection:
           # Use this to specify that a status coming from outside of prow is
           # required.  We use this to require functional jobs running in
           # jenkins are required, for example.
-          required_status_checks:
-            contexts: ["test-v1b1-integration"]
+          branches:
+            main:
+              required_status_checks:
+                contexts: ["test-v1b1-integration"]
```
Comment (smoshiur1237) [inline prow/config/config.yaml:90]: Are we going to add also the `test-v1b1-centos-integration` for bmo PRs?

File: prow/config/config.yaml [hunk 2]
```diff
@@ -84,8 +84,13 @@ branch-protection:
...
+            master:
+              required_status_checks:
+                contexts: ["test-v1b1-integration"]
```
Comment (furkatgofurov7) [inline prow/config/config.yaml:93]: Do  we need this still, I expect we will point all open PRs in BMO to main once we switch to it?

File: prow/config/config.yaml (hunk 1 above)
Comment (furkatgofurov7) [reply prow/config/config.yaml:90]: +1 adding it in the same PR

File: prow/config/config.yaml (hunk 2 above)
Comment (kashifest) [reply prow/config/config.yaml:93]: This has to be removed once main branch is default and all the PRs are pointing to main. In the meanwhile this is needed.

File: prow/config/config.yaml (hunk 1 above)
Comment (kashifest) [reply prow/config/config.yaml:90]: Are we ready yet? Or shall we do this once we have more quota on fra? I dnt have any strong opinion on this. I can add it in the same pr if you folks things its fine

Comment (kashifest) [reply prow/config/config.yaml:90]: Oh wait, we still run integration tests in kna dnt we?

Comment (smoshiur1237) [reply prow/config/config.yaml:90]: yes integration tests are running in kna1. So we shouldn't face any quota issue

Comment (furkatgofurov7) [reply prow/config/config.yaml:90]: just did a quick check, and seems:
- master integration tests are running in Fra1
- PR integration tests in Kna1

Comment (furkatgofurov7) [reply prow/config/config.yaml:90]: So, yes, feature tests, e2e tests, main, master and release* branch tests all are running in Fra, all the remaining ones in Kna.
https://github.com/metal3-io/project-infra/blob/b65db2070e7674acfbaa0a3cb943a7f1252c905f/jenkins/scripts/integration_test.sh#L53-L59

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

File: prow/config/config.yaml (hunk 1 above)
Comment (kashifest) [reply prow/config/config.yaml]: added

File: prow/config/config.yaml [hunk 3]
```diff
@@ -84,8 +84,13 @@ branch-protection:
...
           # jenkins are required, for example.
-          required_status_checks:
-            contexts: ["test-v1b1-integration"]
+          branches:
+            main:
+              required_status_checks:
+                contexts: ["test-v1b1-centos-integration", "test-v1b1-integration"]
+            master:
+              required_status_checks:
+                contexts: ["test-v1b1-centos-integration", "test-v1b1-integration"]
```
Comment (fmuyassarov) [inline prow/config/config.yaml:93]: @kashifest  did we decide that centos job will be mandatory in BMO from now on?

Comment (kashifest) [reply prow/config/config.yaml:93]: We think we can proceed with this. What are your thoughts?

**Pull request merged**